src/esp/assets/ResourceManager.cpp

```diff
@@ -58,6 +63,231 @@ bool ResourceManager::loadScene(const AssetInfo& info,
+          LOG(ERROR) << "Cannot load the mesh, skipping";
+          continue;
+        }
+        meshGroup.push_back(&meshData);
+        
+        /*Magnum::Trade::MeshData3D meshData(
+            reinterpret_cast<std::vector>GMesh->getRenderingBuffer()->mesh)   
+        Magnum::GL::Mesh* mesh  = &meshData->getRenderingBuffer()->mesh;
+        Magnum::GL::Buffer* vbo = &meshData->getRenderingBuffer()->vbo;
+        Magnum::GL::Buffer* cbo = &meshData->getRenderingBuffer()->cbo;
+        Magnum::GL::Buffer* ibo = &meshData->getRenderingBuffer()->ibo;*/
+        //Corrade::Containers::Array<char> v_data = vbo->data();
+        //const std::vector<vec3f> v_data = meshData->get_vbo();
+        //const std::vector<int> i_data   = meshData->get_ibo();
+        // TODO (JH): need this in order to get Replica mesh working
+        //Magnum::MeshTools::transformPointsInPlace(transform,
+        //                                          meshData->positions(0));
+      }
+
+      // GLB Mesh
+      else if (info.type == AssetType::MP3D_MESH) {
+      
+        //std::vector<std::shared_ptr<BaseMesh>> sceneMesh(&meshes_[start], &meshes[end]);
+        // Apply in-place transformation to collision mesh, to be consistent with
+        //  display mesh
+        //Magnum::Trade::MeshData3D* meshData;
+        quatf quatf = quatf::FromTwoVectors(info.frame.front(), geo::ESP_FRONT);
+        Magnum::Quaternion quat = Magnum::Quaternion(quatf);
+        GltfMeshData* gltfMeshData = dynamic_cast<GltfMeshData*>(meshes_[mesh_i].get());
+        auto& meshData = gltfMeshData->getMeshData();
+        if (!meshData) {
+          LOG(ERROR) << "Cannot load the mesh, skipping";
+          continue;
+        }
+        Magnum::Matrix4 transform =
+            Magnum::Matrix4::rotation(quat.angle(), quat.axis().normalized());
+        Magnum::MeshTools::transformPointsInPlace(transform,
+                                                  meshData->positions(0));
+        meshGroup.push_back(&meshData);
+        /*LOG(INFO) << "Initializing scene";
+        LOG(INFO) << "Scene built in angle " << float(quat.angle());
+        LOG(INFO) << "Scene built in axis " << Eigen::Map<vec3f>(quat.axis().data());*/
+      }
+
+    }
+
+    bool sceneSuccess = _physicsManager.initScene(
+        info, metaData, meshGroup, physNode);
+    LOG(INFO) << "Created mesh object";
+    if (!sceneSuccess) {
+      LOG(INFO) << "Physics manager failed to initialize object";
+      return false;
+    }
+
+    else {
+      return false;
+    }
+  }
+
+  return meshSuccess;
+}
+
+bool ResourceManager::loadObject(const AssetInfo& info,
+                                 PhysicsManager& _physicsManager,
+                                 scene::SceneNode* parent,
+                                 bool attach_physics, /* = false */
+                                 DrawableGroup* drawables /* = nullptr */,
+                                 physics::BulletRigidObject** physNode) {
```

Comment:
This hack that we did to get it working is scary and shouldn't get to live :).  Returning an Optional<scene::SceneNode*> is the way to do this correctly.  `attach_physics` will get to decided if the returned node is a pointer to a `physics::BulletRigidObject` that was down-casted or a pointer to a SceneNode.